app/assets/javascripts/preferences/panes/account/changePassword/index.tsx

```diff
@@ -0,0 +1,192 @@
+    const success = await application.validateAccountPassword(currentPassword);
+    if (!success) {
+      applicationAlertService.alert(
+        'The current password you entered is not correct. Please try again.'
+      );
+    }
+    return success;
+  };
+
+  const resetProgressState = () => {
+    setSubmitButtonTitle(SubmitButtonTitles.Default);
+    setIsContinuing(false);
+  };
+
+  const processPasswordChange = async () => {
+    await application.downloadBackup();
+
+    setLockContinue(true);
+
+    const response = await application.changePassword(
+      currentPassword,
+      newPassword
+    );
+
+    const success = !response.error;
+
+    setLockContinue(false);
+
+    return success;
+  };
+
+  const dismiss = () => {
+    if (lockContinue) {
+      applicationAlertService.alert(
+        'Cannot close window until pending tasks are complete.'
+      );
+    } else {
+      onCloseDialog();
+    }
+  };
+
+  const handleSubmit = async () => {
+    if (lockContinue || isContinuing) {
+      return;
+    }
+
+    if (currentStep === Steps.FinishStep) {
+      dismiss();
+      return;
+    }
+    setIsContinuing(true);
+    setSubmitButtonTitle(SubmitButtonTitles.GeneratingKeys);
+
+    const valid = await validateCurrentPassword();
+
+    if (!valid) {
+      resetProgressState();
+      return;
+    }
+
+    const success = await processPasswordChange();
+    if (!success) {
+      resetProgressState();
+      return;
+    }
+    setIsContinuing(false);
+    setSubmitButtonTitle(SubmitButtonTitles.Finish);
+    setCurrentStep(Steps.FinishStep);
+  };
+
+  const handleDialogClose = () => {
+    if (lockContinue) {
+      applicationAlertService.alert(
+        'Cannot close window until pending tasks are complete.'
+      );
+    } else {
+      onCloseDialog();
+    }
+  };
+
+  return (
+    <div>
+      <ModalDialog>
+        <ModalDialogLabel closeDialog={handleDialogClose}>
+          Change Password
+        </ModalDialogLabel>
+        <ModalDialogDescription>
+          {currentStep === Steps.InitialStep && (
+            <ChangePasswordForm
+              setCurrentPassword={setCurrentPassword}
+              setNewPassword={setNewPassword}
+              setNewPasswordConfirmation={setNewPasswordConfirmation}
+            />
+          )}
+          {currentStep === Steps.FinishStep && <ChangePasswordSuccess />}
+        </ModalDialogDescription>
+        <ModalDialogButtons>
+          {currentStep === Steps.InitialStep && (
+            <Button
+              className='min-w-20'
+              type='normal'
+              label='Cancel'
+              onClick={handleDialogClose}
+            />
+          )}
+          <Button
+            className='min-w-20'
+            type='primary'
+            label={submitButtonTitle}
+            onClick={handleSubmit}
+          />
+        </ModalDialogButtons>
+      </ModalDialog>
+    </div>
+  );
+};
+
+export default ChangePassword;
```

Comment:
Why? :) While it's codebase-related and I've seen and used both approaches, in `web` the (most of) components are exported with `default` approach. I'm ok with both named and default exports, but since "default" approach seems to be most wide-spread approach in "web" repo, I kept using it.

Reply:
Because, it doesn't foster importing Components under the same name that they were exported as and it will need additional refactors if other types were exported, such as the component's props.

Reply:
I think I may have originally introduced `export default ...` a while back, and I don't necessarily stand by it :) I do think named exports are a bit more conventional and consistent with how we do things now in other codebases.

Reply:
Usually IDEs do that job pretty well, and even if we rename the component exported by "default" way, they do all the required renaming and correct importing. As I said above - I'm ok with both approaches, and as default exports were in our codebase when I started working on it, I went that way to keep consistency in our codebase.

Reply:
In any case, I think it's a good time to decide what way we want to continue. Neither way is "more correct" than another actually, both approaches are correct.

Reply:
I'd say let's go with named exports. SNJS code is far newer than web code and has almost no default exports, so we could aim in that direction for consistency.

Reply:
Well, if we want to have consistency then we shouldn't use default exports because they don't allow you to export multiple types.

Reply:
Default exports are mostly used for components. Types and other stuff like util functions are exported by names.
I updated exports in this PR to named ones.